pydantic/color.py

```diff
@@ -20,21 +20,28 @@
 if TYPE_CHECKING:  # pragma: no cover
     from .types import CallableGenerator
 
-
 ColorTuple = Union[Tuple[int, int, int], Tuple[int, int, int, float]]
 ColorType = Union[ColorTuple, str]
 HslColorTuple = Union[Tuple[float, float, float], Tuple[float, float, float, float]]
 
 
-class RGBA(NamedTuple):
+class RGBA:
     """
     Internal use only as a representation of a color.
     """
 
-    r: float
-    g: float
-    b: float
-    alpha: Optional[float]
+    __slots__ = 'r', 'g', 'b', 'alpha', '_tuple'
```

Comment:
I'm not sure about the impact of `__slots__` here, but it felt more in the spirit of the tuple it replaced. I can remove it if desired.